app/server.go

```diff
@@ -484,6 +484,8 @@ func NewServer(options ...Option) (*Server, error) {
 		pluginsEnvironment.InitPluginHealthCheckJob(*s.Config().PluginSettings.Enable && *s.Config().PluginSettings.EnableHealthCheck)
 	}
 	s.AddConfigListener(func(_, c *model.Config) {
+		s.PluginsLock.RLock()
+		defer s.PluginsLock.RUnlock()
```

Comment:
It looks safe to me but we could also avoid the `defer` since we only need to lock to read the env from the server struct.

Reply:
makes sense to me